src/operator/nn/upsampling-inl.h

```diff
@@ -96,28 +98,51 @@ void UpSamplingForward(const OpContext &ctx, const UpSamplingParam &param,
   if (req[up_enum::kOut] == kNullOp) {
     return;
   }
+
   Stream<xpu> *s = ctx.get_stream<xpu>();
   Tensor<xpu, 4, DType> out = out_data[up_enum::kOut].get<xpu, 4, DType>(s);
   if (param.num_args > 1) {
     int begin = 0;
     for (int i = 0; i < param.num_args; ++i) {
       Tensor<xpu, 4, DType> data = in_data[i].get<xpu, 4, DType>(s);
       int end = begin + data.size(1);
-      int scale = out_data[up_enum::kOut].size(2)/in_data[i].size(2);
+      
+      int scale_h = out_data[up_enum::kOut].size(2)/in_data[i].size(2);//3rd dimension of TBlob (2nd from 4th dimension)
+      int scale_w = out_data[up_enum::kOut].size(1)/in_data[i].size(1);//4th dimension of TBlob (1st from 4th dimension)
+      
       if (param.multi_input_mode == up_enum::kSum) {
         if (i == 0) {
-          Assign(out, req[up_enum::kOut], upsampling_nearest(data, scale));
+
+          Assign(out, req[up_enum::kOut], upsampling_nearest(data, scale_h, scale_w));
+
         } else {
-          out += upsampling_nearest(data, scale);
+
+          out += upsampling_nearest(data, scale_h, scale_w);
+
         }
       } else {
-        Assign(slice<1>(out, begin, end), req[up_enum::kOut], upsampling_nearest(data, scale));
+
+        Assign(slice<1>(out, begin, end), req[up_enum::kOut], upsampling_nearest(data, scale_h, scale_w));
+
       }
       begin = end;
     }
   } else {
     Tensor<xpu, 4, DType> data = in_data[up_enum::kData].get<xpu, 4, DType>(s);
-    Assign(out, req[up_enum::kOut], upsampling_nearest(data, param.scale));
+
+    int scale_h, scale_w;
+    if (param.scale.ndim() == 1) {
+      scale_h = param.scale[0];
+      scale_w = param.scale[0];
+    } else if (param.scale.ndim() == 2) {
+      scale_h = param.scale[0];
+      scale_w = param.scale[1];
+    } else if (param.scale.ndim() == 4) {
+      scale_h = param.scale[2];
+      scale_w = param.scale[3];
+    }
```

Comment:
can we move this if statement block to a common function or store the result of this computation? its referenced in multiple places.